File: find_draconians.py

import json

try:
    with open("src/lib/data/monsters_2024_final.json", "r", encoding="utf-8") as f:
        # Read line by line to get line numbers
        for i, line in enumerate(f):
            if "Draconian" in line and "name" in line:
                print(f"Line {i+1}: {line.strip()}")
except Exception as e:
    print(f"Error: {e}")

File: find_draconians.js

const fs = require('fs');
const readline = require('readline');

async function processLineByLine() {
    const fileStream = fs.createReadStream('src/lib/data/monsters_2024_final.json');

    const rl = readline.createInterface({
        input: fileStream,
        crlfDelay: Infinity
    });

    let lineNumber = 0;
    for await (const line of rl) {
        lineNumber++;
        if (line.includes('Draconian') && line.includes('name')) {
            console.log(`Line ${lineNumber}: ${line.trim()}`);
        }
    }
}

processLineByLine();
